modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js

```diff
@@ -12,96 +12,79 @@
  * details.
  */
 
-import {Editor} from 'frontend-editor-ckeditor-web';
-import React from 'react';
+import {ClassicEditor} from 'frontend-editor-ckeditor-web';
+import React, {useMemo} from 'react';
 
 import {FieldBase} from '../FieldBase/ReactFieldBase.es';
 import {useSyncValue} from '../hooks/useSyncValue.es';
 
 const CKEDITOR_CONFIG = {
+	autoGrow_bottomSpace: 50,
+	autoGrow_maxHeight: 600,
+	autoGrow_minHeight: 215,
+	resize_enabled: true,
 	toolbar: [
-		{items: ['Undo', 'Redo'], name: 'clipboard'},
-		'/',
-		{
-			items: [
-				'Bold',
-				'Italic',
-				'Underline',
-				'Strike',
-				'-',
-				'CopyFormatting',
-				'RemoveFormat',
-			],
-			name: 'basicstyles',
-		},
-		{
-			items: [
-				'NumberedList',
-				'BulletedList',
-				'-',
-				'Outdent',
-				'Indent',
-				'-',
-				'Blockquote',
-				'-',
-				'JustifyLeft',
-				'JustifyCenter',
-				'JustifyRight',
-				'JustifyBlock',
-			],
-			name: 'paragraph',
-		},
-		{items: ['Link', 'Unlink', 'Anchor'], name: 'links'},
-		{
-			items: ['Image', 'Table', 'HorizontalRule', 'SpecialChar'],
-			name: 'insert',
-		},
-		'/',
-		{items: ['Styles', 'Format', 'Font', 'FontSize'], name: 'styles'},
-		{items: ['TextColor', 'BGColor'], name: 'colors'},
-		{items: ['Maximize'], name: 'tools'},
-		{
-			items: ['Source'],
-			name: 'document',
-		},
-	],
+		['Undo', 'Redo'],
+		['Styles'],
+		['Bold', 'Italic', 'Underline'],
+		['RemoveFormat'],
+		['NumberedList', 'BulletedList'],
+		['Blockquote'],
+		['Link', 'Unlink'],
+		['Table', 'Image', 'VideoEmbed'],
+		['Source'],
+	]
 };
 
 const RichText = ({
+	editorConfig,
 	id,
 	name,
 	onChange,
 	predefinedValue,
 	readOnly,
 	value,
+	visible,
 	...otherProps
 }) => {
 	const [currentValue, setCurrentValue] = useSyncValue(
 		value ? value : predefinedValue
 	);
 
-	const editorProps = {
-		config: CKEDITOR_CONFIG,
-		data: currentValue,
-	};
+	const normalizedEditorConfig = useMemo(() => {
+		if (!editorConfig) {
+			return {};
+		}
 
-	if (readOnly) {
-		editorProps.readOnly = true;
-		editorProps.style = {pointerEvents: 'none'};
-	}
-	else {
-		editorProps.onChange = (event) => {
-			const newValue = event.editor.getData();
-
-			setCurrentValue(newValue);
+		const config =
+			editorConfig.editorConfig.JSONObject ??
+			editorConfig.editorConfig;
 
-			onChange(event, newValue);
+		return {
+			...config,
+			...CKEDITOR_CONFIG,
```

Comment:
Won't `CKEDITOR_CONFIG` override `config`? I'd expect `CKEDITOR_CONFIG` to be a sensible default, which is mostly already provided by `ClassicEditor`...

Reply:
The `CKEDITOR_CONFIG` here is just for toolbar and autogrow customizations, I can add this to the field in Java if you wish.